File: test/create_test_pdf.py
"""
Create a simple test PDF for testing page numbers and protection features
"""

from reportlab.pdfgen import canvas
from reportlab.lib.pagesizes import letter
import os

def create_test_pdf(filename, num_pages=5):
    """Create a multi-page test PDF"""
    c = canvas.Canvas(filename, pagesize=letter)
    width, height = letter
    
    for page_num in range(1, num_pages + 1):
        # Add some content to each page
        c.setFont("Helvetica-Bold", 24)
        c.drawString(100, height - 100, f"Test Page {page_num}")
        
        c.setFont("Helvetica", 12)
        c.drawString(100, height - 150, f"This is page {page_num} of {num_pages}")
        c.drawString(100, height - 180, "This PDF is for testing:")
        c.drawString(100, height - 200, "• Add Page Numbers feature")
        c.drawString(100, height - 220, "• Protect PDF feature")
        
        # Add page footer
        c.setFont("Helvetica-Oblique", 10)
        c.drawString(100, 50, "Test PDF created for feature testing")
        
        c.showPage()
    
    c.save()
    print(f"✅ Created {filename} with {num_pages} pages")

if __name__ == "__main__":
    # Create test directory if it doesn't exist
    os.makedirs("test_pdfs", exist_ok=True)
    
    # Create test PDFs
    create_test_pdf("test_pdfs/test_5_pages.pdf", 5)
    create_test_pdf("test_pdfs/test_3_pages.pdf", 3)
    
    print("\n✅ Test PDFs created successfully!")
    print("📁 Files saved in: test_pdfs/")

File: test/create_test_pdf.js
/**
 * Create test PDF files for testing features
 */

const { PDFDocument, rgb, StandardFonts } = require('pdf-lib');
const fs = require('fs').promises;
const path = require('path');

async function createTestPDF(filename, numPages) {
  const pdfDoc = await PDFDocument.create();
  const font = await pdfDoc.embedFont(StandardFonts.Helvetica);
  const boldFont = await pdfDoc.embedFont(StandardFonts.HelveticaBold);

  for (let i = 1; i <= numPages; i++) {
    const page = pdfDoc.addPage([612, 792]); // Letter size
    const { width, height } = page.getSize();

    // Title
    page.drawText(`Test Page ${i}`, {
      x: 100,
      y: height - 100,
      size: 24,
      font: boldFont,
      color: rgb(0, 0, 0),
    });

    // Subtitle
    page.drawText(`This is page ${i} of ${numPages}`, {
      x: 100,
      y: height - 150,
      size: 12,
      font,
      color: rgb(0, 0, 0),
    });

    // Content
    page.drawText('This PDF is for testing:', {
      x: 100,
      y: height - 180,
      size: 12,
      font,
      color: rgb(0, 0, 0),
    });

    page.drawText('• Add Page Numbers feature', {
      x: 100,
      y: height - 200,
      size: 12,
      font,
      color: rgb(0, 0, 0),
    });

    page.drawText('• Protect PDF feature', {
      x: 100,
      y: height - 220,
      size: 12,
      font,
      color: rgb(0, 0, 0),
    });

    // Footer
    page.drawText('Test PDF created for feature testing', {
      x: 100,
      y: 50,
      size: 10,
      font,
      color: rgb(0.5, 0.5, 0.5),
    });
  }

  const pdfBytes = await pdfDoc.save();
  await fs.writeFile(filename, pdfBytes);
  console.log(`✅ Created ${filename} with ${numPages} pages`);
}

async function main() {
  try {
    // Create test directory
    const testDir = path.join(__dirname, '..', 'test_pdfs');
    await fs.mkdir(testDir, { recursive: true });

    // Create test PDFs
    await createTestPDF(path.join(testDir, 'test_5_pages.pdf'), 5);
    await createTestPDF(path.join(testDir, 'test_3_pages.pdf'), 3);

    console.log('\n✅ Test PDFs created successfully!');
    console.log('📁 Files saved in: test_pdfs/');
  } catch (error) {
    console.error('Error creating test PDFs:', error);
    process.exit(1);
  }
}

main();
